src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Usage/DoNotCheckFlagFromDifferentEnum.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.NetCore.Analyzers.Usage
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public sealed class DoNotCheckFlagFromDifferentEnum : DiagnosticAnalyzer
```

Comment:
I don't really like this class name. Any suggestion?

Reply:
How about `ProvideCorrectArgumentToEnumHasFlag`?

Reply:
I think I like `ProvideCorrectArgumentToEnumHasFlag` more as it gives more info about the context.